addons/account/migrations/12.0.1.1/post-migration.py

```diff
@@ -0,0 +1,162 @@
+# Copyright 2018 Eficent <http://www.eficent.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+from psycopg2.extensions import AsIs
+
+
+def map_account_payment_term_line_option(cr):
+    openupgrade.map_values(
+        cr,
+        openupgrade.get_legacy_name('option'),
+        'option',
+        [('fix_day_following_month', 'after_invoice_month'),
+         ('last_day_following_month', 'day_following_month'),
+         ('last_day_current_month', 'day_current_month'),
+         ],
+        table='account_payment_term_line', write='sql')
+
+
+def map_account_tax_type_tax_use(cr):
+    cr.execute(
+        """
+        UPDATE account_tax
+        SET type_tax_use = 'adjustment'
+        WHERE type_tax_use = 'none' AND %s = TRUE
+        """, (AsIs(openupgrade.get_legacy_name('tax_adjustment')), ),
+    )
+    cr.execute(
+        """
+        UPDATE account_tax_template
+        SET type_tax_use = 'adjustment'
+        WHERE type_tax_use = 'none' AND %s = TRUE
+        """, (AsIs(openupgrade.get_legacy_name('tax_adjustment')), ),
+    )
+
+
+def fill_bank_statement_line_account_number(cr):
+    cr.execute(
+        """
+        UPDATE account_bank_statement_line absl
+        SET account_number = rpb.acc_number
+        FROM res_partner_bank rpb
+        WHERE absl.bank_account_id = rpb.id AND rpb.acc_number IS NOT NULL
+        """
+    )
+
+
+def fill_account_chart_template_account_code_prefix(cr):
+    cr.execute(
+        """
+        UPDATE account_chart_template act
+        SET bank_account_code_prefix = rc.bank_account_code_prefix
+        FROM res_company rc
+        WHERE act.%s = rc.id AND act.bank_account_code_prefix IS NULL
+        """, (AsIs(openupgrade.get_legacy_name('company_id')), ),
+    )
+    cr.execute(
+        """
+        UPDATE account_chart_template act
+        SET cash_account_code_prefix = rc.cash_account_code_prefix
+        FROM res_company rc
+        WHERE act.%s = rc.id AND act.cash_account_code_prefix IS NULL
+        """, (AsIs(openupgrade.get_legacy_name('company_id')), ),
+    )
+    # we can't use here new transfer_account_code_prefix field of res_company:
+    cr.execute(
+        """
+        UPDATE account_chart_template act
+        SET transfer_account_code_prefix = aa.code
+        FROM res_company rc
+        LEFT JOIN account_account aa ON rc.transfer_account_id = aa.id
+        WHERE act.%s = rc.id
+        """, (AsIs(openupgrade.get_legacy_name('company_id')), ),
+    )
+
+
+def fill_account_payment_partner_bank_account_id(cr):
+    cr.execute(
+        """
+        UPDATE account_payment ap
+        SET partner_bank_account_id = COALESCE(rp_rpb_rel.rpb, rp_rpb_rel2.rpb)
+        FROM res_partner rp
+        LEFT JOIN (
```

Comment:
This part can be removed, as bank account are only allowed to be on commercial partners.

Reply:
Which part exactly? I thought this superquery was perfect 😅

Reply:
This first LEFT JOIN. It's not needed if you think twice :wink:

Reply:
After more analysis, I checked that we should not fill this optional field. Odoo makes you a proposal, but not forces you to put a bank account unless if exists a payment method that forces it. But in odoo community there doesn't exist any of this kind of methods as you can see that `_get_method_codes_using_bank_account` method is an empty list.

Reply:
I saw the same, but also don't discard it as a community module can use it, but yes, for me it's OK to remove it. But please document all of this in the analysis file.